packages/firebase_core/lib/firebase_core_tizen.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 Invertase Limited. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file.
+
+library firebase_core_desktop;
+
+import 'package:firebase_core_dart/firebase_core_dart.dart' as core_dart;
+import 'package:firebase_core_platform_interface/firebase_core_platform_interface.dart';
+
+part 'firebase_app_desktop.dart';
+
+/// Desktop implementation of FirebaseCore for managing Firebase app
+/// instances.
+class FirebaseCore extends FirebasePlatform {
+  /// Called by PluginRegistry to register this plugin as the implementation for Desktop
+  static void registerWith() {
+    FirebasePlatform.instance = FirebaseCore();
+  }
+
+  // ignore: public_member_api_docs
+  static void register() => registerWith();
```

Comment:
`registerWith` is for desktop plugins. Probably you don't need both but only need `register`.

```suggestion
  /// Registers this class as the default instance of [FirebasePlatform].
  static void register() {
    FirebasePlatform.instance = FirebaseCore();
  }
```

Reply:
I wanted to minimize the changes in this file, but adding 4 letters is actually smaller than adding 2 lines.